### PR TITLE
Add the makeBuffer interface to lz4js

### PR DESCRIPTION
**Pull request by hu0529**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by hu0529** [comment]
Add the makeBuffer interface to lz4js

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/lz4js/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4ad8da8363392fc2ce07a938644133378c653b95
</p>

**Comment by hu0529** [comment]
> ## Formatting
> The following files are not formatted:
> 
> 1. types/lz4js/index.d.ts
> 
> Consider running `pnpm dprint fmt` on these files to make review easier.
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [4ad8da8](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/4ad8da8363392fc2ce07a938644133378c653b95)

Running pnpm dprint fmt has been executed

File: types/lz4js/index.d.ts [hunk 1]
```diff
@@ -10,19 +10,21 @@ export function compress(src: Uint8Array | number[], maxSize?: number): Uint8Arr
 export function decompress(src: Uint8Array | number[], maxSize?: number): Uint8Array;
 
 export function compressBlock(
-    src: Uint8Array | number[],
-    dst: Uint8Array | number[],
-    sIndex: number,
```
Comment (RyanCavanaugh) [inline types/lz4js/index.d.ts:15]: Undo unrelated formatting change

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
Blocking PRs from both authors pending https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69172#issuecomment-2035417345

**Review by peterblazejewicz — APPROVED** [review]

**Comment by hu0529** [comment]
Ready to merge

**Pull request merged**